refactor: new documentation & formatting rules

xnacly opened the pull request:
## Abstract:
This pull request serves the purpose of improving the documentation and the workflow of adding examples, comments, and descriptions by moving away from manually writing HTML documentation to [jsdoc](https://github.com/jsdoc/jsdoc) and generating the documentation using [typedoc](https://typedoc.org/).

### Infos:
#### How to generate the new documentation:
```
npm i -g typedoc
git clone https://github.com/trantlabs/missing-native-js-functions mnjsf
cd mnjsf
typedoc src/*.ts
```
The generated documentation can be found in `./docs`

## Screenshots of the generated documentation
### Readme:
![grafik](https://user-images.githubusercontent.com/47723417/208883078-f30ac727-b464-42ab-8976-5d517af5268a.png)
### Description and Comments for methods:
#### Screenshot
![grafik](https://user-images.githubusercontent.com/47723417/208883143-4faf4c8d-8841-4347-89e8-1723e5dc0f60.png)
#### Corresponding jsdoc function comments:
![grafik](https://user-images.githubusercontent.com/47723417/208883548-e0977837-d799-4382-a7af-9547aeafd38f.png)

## Fixed bugs:
While writing these comments I came across some bugs which I promptly fixed:
- wrong return type for `String.equalsIgnoreCase()`, changed from `string` to `boolean` https://github.com/xNaCly/missing-native-js-functions/commit/741ea9b5cf04633db03898725b6b5ec6f1379729 
- `Date.nowSeconds()` is not a function, constructed Date before calling `.nowSeconds()` on it https://github.com/xNaCly/missing-native-js-functions/commit/7fbaf97338ee8f55538140ef04f54c5caae8ad84
- add missing `timezone: string` function parameter to `DateConstructor.setTimeZone()` https://github.com/xNaCly/missing-native-js-functions/commit/1b44edde000c49759280f2ad15af935458ce152d
- fix typo in `DateConstructor.setTimeZone()` which made the function always return the local date https://github.com/xNaCly/missing-native-js-functions/commit/74b02456f7e45d8a6998718eff3f1c230b860635

## List of changes:
- removed old documentation
- fixed some bugs
- wrote jsdoc and examples for:
  - [x] Array.ts
  - [x] Date.ts
  - [x] Global.ts
  - [x] Math.ts
  - [x] Number.ts
  - [x] Object.ts
  - [x] Promise.ts
  - [x] String.ts
- removed documentation from Readme

## Future todo:
- write tests
- add workflow to build and publish documentation
- add ci

xnacly committed:
doc: add docs for Globals

xnacly committed:
doc: add docs for Date

xnacly committed:
fix: doc for Global.sleep()

xnacly committed:
fix: add prettierrc, format whole project

xnacly committed:
doc: remove documentation from readme, fix issues

xnacly committed:
doc: remove old documentation

xnacly committed:
docs: Array

Intevel commented on README.md:
```diff
@@ -7,20 +6,19 @@
+## Why use mnjsf?
 
-mnjsf delivers some functions which are missing in JavaScript
+mnjsf delivers functions missing from JavaScript
```
Maybe add a `native` before `JavaScript`?

xnacly committed:
chore: add documentation dir to gitignore

xnacly committed:
fix: return type for String.equalsIgnoreCase

changed return type from string to boolean

xnacly committed:
doc: String.ts, Promise.ts

add docs for String methods:
- capitalize
- replaceAll
- similarity
- join
- partition
- toNumber
- toBigInt
- equalsIgnoreCase
- count
- swapcase
- title
- toObject
- toBoolean
add docs for Promise.caught

xnacly committed:
doc: add examples to Global.ts and Promise.ts

xnacly committed:
fix: Date.nowSeconds is not a function

constructed Date before calling .nowSeconds on it

xnacly committed:
fix: missing function parameter Date.setTimezone

`timezone: string` was missing

xnacly committed:
doc: add Date.ts

xNaCly commented:
## Switching Interfaces
Im currently thinking of switching to the `Date` interface, away from the `DateConstructor` interface, due to the fact, that i wasn't able to call the defined functions on the Date without creating an instance.:
![image](https://user-images.githubusercontent.com/47723417/208902560-53ab1a53-733a-4b05-9080-c19a755b7189.png)

I also wasn't able to see any doc comments on these functions. It also doesn't make sense to define functions which rely on an instance, such as `setInstance` and `isInPast` on a constructor.

## Fixing bugs
Also, the `DateConstructor.setTimezone` function does not work as intended, it always returns a local date:
```ts
	setTimezone: function (timezone: string) {
		return new Date(
			this.toLocaleString("en-US", {
				timezone,
			})
		);
	},
```
According to the docs: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DateTimeFormat/DateTimeFormat#options we should be able to fix this by replacing `timezone` in the options with `timeZone`.

Test in the browser dev tools:
![image](https://user-images.githubusercontent.com/47723417/208901938-a8028dd7-9edc-4734-94fb-89b8a6ac43e2.png)

@SamuelScheit @Intevel

xnacly committed:
fix: typo in setTimezone

- timezone -> timeZone
due to the typo in timezone, the function always returned the local time.

Docs: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DateTimeFormat/DateTimeFormat#options

xnacly committed:
doc: add examples, finish docs

xnacly committed:
doc: remove buggy doc comments

xnacly committed:
feat: fix Date and Dateconstructor interfaces

- implement nowSeconds on the Date object
- move setTimezone, isInPast implementation to the Date object
- move setTimezone, isInPast definition to the Date interface
- define nowSeconds on the DateConstructor interface

xnacly committed:
doc: add Array.findMap docs

Intevel commented on src/Array.ts:
```diff
@@ -1,148 +1,262 @@
+		if (index === -1) return this;
+
+		this.splice(index, 1);
+		return this; // .filter((e) => e !== elem);
```
Remove comment

Intevel commented on src/Object.ts:
```diff
@@ -1,71 +1,112 @@
-  }
-})
+		Object.keys(this).forEach((key) => {
+			// @ts-ignore
```
Use ts-expect-error instead of ts-ignore

xnacly committed:
fix: make requested changes from @Intevel

- Use ts-expect-error instead of ts-ignore
- Remove comment
